### ISSUE TITLE
RequireThis false positive on method reference 

### ISSUE THREAD

**Issue by artembilan** [checkstyle/checkstyle#7306]
```
/var/tmp $ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
    <module name="RequireThis">
      <property name="validateOnlyOverlapping" value="false"/>
	</module>
  </module>
</module>

/var/tmp $ cat RequireThisRegression.java
import java.util.ArrayList;
import java.util.List;

public class RequireThisRegression {

    List<String> add = new ArrayList<>();

    void someMethod() {
        List<String> test = new ArrayList<>();
        test.forEach(this.add::add);
    }

}

/var/tmp $ java -jar checkstyle-8.27-all.jar -c config.xml RequireThisRegression.java
Starting audit...
[ERROR] /var/tmp/RequireThisRegression.java:10:32: Reference to instance variable 'add' needs "this.". [RequireThis]
Audit done.
Checkstyle ends with 1 errors.
```

The point is if we disable `validateOnlyOverlapping` and have a property with the same name as some method we would like to use via method reference, the `RequireThis` check fails requesting a `this.` prefix for that method name instead of understanding that this one is a method reference.
Works well with the previous `8.26` version.

Thanks

**Comment by romani** [comment]
@strkkk , is resulted by update to fix https://github.com/checkstyle/checkstyle/issues/7066 , please step in.

@artembilan , looks like your actively using this Check - https://github.com/checkstyle/checkstyle/issues?q=is%3Aissue+author%3Aartembilan+is%3Aclosed , thanks a lot for your prompt feedback. 
Do you have any opensource project where such Check is enforced to let us use it in regression testing ... it is not easy Check ... and we do not want to make extra regressions.

**Comment by artembilan** [comment]
Hello, @romani, and thank you for prompt response!

Yep, I do open source project: https://github.com/spring-projects/spring-integration.

I know that this check is not so important in modern IDE, plus we use SonarQube to be sure that local variables don't clash with properties, but it still really helps to review pull requests from GitHub.

**Comment by romani** [comment]
please share details on how to run checkstyle on spring and get a build failure (error severity), 

with such details I will add project to our regression and you will forget about problems with regression from checkstyle.  All I need is that chekcstyle is executed by some command line and it fails when at least one violation happening.

**Comment by artembilan** [comment]
We use Gradle in the project, so a command for you to run Checkstyle is like this `./gradlew clean check --parallel -x test`.

Let me know if that help!

I don't argue that we don't use all the checks in the project to fully cover you (a lot of JavaDocs need to be added 😄 ), but at least something what can help you stay clean as much as possible.

Very appreciate!

By the way: we have a task somewhere in the `build.gradle` like this:
```
checkstyle {
	configFile = file("$rootDir/src/checkstyle/checkstyle.xml")
	toolVersion = '8.26'
}
```

Let me know if you need to have it somehow configurable externally! For example the version could be as a command line arg or we can expose it into the `gradle.properties`.

I'm not sure if you are going to contribute something back into my project, but at least automatic upgrade to a new version would be great 😄 
Anyway: different story. 
Right now let me know what can I do from my project perspective to simplify your life.

Thank you again!

**Comment by romani** [comment]
>Let me know if you need to have it somehow configurable externally!

Yes, please, we will run on snapshot version. 
Alternatively you put checkstyle version to some distinctive property to let send replace version reliably by sed.

**Comment by strkkk** [comment]
> @strkkk , is resulted by update to fix #7066 , please step in.

I already made a fix, will submit PR soon

**Comment by artembilan** [comment]
Hey, @romani !

See the commit in my project for what we have discussed: https://github.com/spring-projects/spring-integration/commit/2e10cc8e9ce903af358a3faf6cbd38028182e465

So, to fail it locally I use this command `./gradlew clean check --parallel -x test -PcheckstyleVersion=8.27`.

Yes, our `master` is really a source of truth for Checkstyle: we have a `check` task integration into our project build. We don't allow PR to make it into the `master` until Travis is OK with the change.
Also we build the master before publishing on Bamboo with the `check` task as well.
So, you really can be sure to fork our repo and pull the latest changes from the `upstream master` for your CI goals.

@strkkk , thank you very much for the fix!

Let's see if my project let it go when you integration it into your CI pipe-line.

Thank you all very much for such a great open source cooperation!

**Comment by pbludov** [comment]
fix is merged.
